src/metabase/search/scoring.clj

```diff
@@ -117,6 +133,18 @@
   [[score-1 _result-1] [score-2 _result-2]]
   (compare score-1 score-2))
 
+(defn- serialize
+  "Massage the raw result from the DB into something more useful for the client"
+  [{:keys [collection_id collection_name name display_name] :as row} {:keys [score match column match-context] :as hit}]
+  (-> row
+      (assoc
+       :name           (if (or (= column :name) (nil? display_name)) name display_name)
+       :matched_column column
+       :matched_text   match
+       :context        (when-not (#{:name :display_name :collection_name} column) match-context) ;; TODO pull this out somewhere more responsible
```

Comment:
these "magic" column names should live in `search.config` and both the `when-not` and `if` (L141) are a little jank, but I think the surrounding logic is going to change in the near future and don't want to waste cycles overoptimizing